PolyEngine/Engine/Src/Engine.cpp

```
@@ -23,46 +23,28 @@ void Engine::Deinit()
 	Renderer = nullptr;
 }
 
-void Engine::RegisterPreupdatePhase(PhaseUpdateFunction phaseFunction)
+void Engine::RegisterUpdatePhase(const PhaseUpdateFunction& phaseFunction, eUpdatePhaseOrder order)
 {
-	for (auto& iter : GamePreupdatePhases)
+	HEAVY_ASSERTE(order != eUpdatePhaseOrder::_Count, "_Count enum value passed to RegisterUpdatePhase(), which is an invalid value");
+	Dynarray<PhaseUpdateFunction>& UpdatePhases = GameUpdatePhases[static_cast<int>(order)];
+	for (auto& iter : UpdatePhases)
 	{
 		if (iter.target<PhaseUpdateFunction>() == phaseFunction.target<PhaseUpdateFunction>())
```

This should be an assertion. We don't want it to silently fail.